File: user-service/src/main/java/rs/edu/raf/service/UserService.java
package rs.edu.raf.service;

import rs.edu.raf.dto.*;

public interface UserService {

    /**
     * Log user into account
     * @param loginDto user credentials
     * @return {@link LoginResponseDTO} if login is successful
     */
    LoginResponseDTO loginUser(LoginDto loginDto);

    /**
     * Gets the currently logged user
     * @return {@link LoggedUserDTO} object containing information
     * about currently logged user
     */
    LoggedUserDTO getLoggedUser();

    /**
     * Changes the password of a user.
     *
     * @param changePasswordDTO   The DTO containing information to change the password.
     * @return True if the password is changed
     */
    boolean changeUserPassword(ChangePasswordDTO changePasswordDTO);

    /**
     * Changes the password of a user using a code.
     *
     * @param changePasswordWithCodeDTO The DTO containing information to change the password.
     * @return True if password is changed
     */
    boolean changeUserPasswordWithCode(ChangePasswordWithCodeDTO changePasswordWithCodeDTO);
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/service/UserService.kt
package org.example.userservicekotlin.service

import org.example.userservicekotlin.dto.*

interface UserService {
    /**
     * Log user into account
     * @param loginDto user credentials
     * @return [LoginResponseDTO] if login is successful
     */
    fun loginUser(loginDto: LoginDTO): LoginResponseDTO

    /**
     * Gets the currently logged user
     * @return [LoggedUserDTO] object containing information
     * about currently logged user
     */
    fun getLoggedUser(): LoggedUserDTO

    /**
     * Changes the password of a user.
     *
     * @param changePasswordDTO   The DTO containing information to change the password.
     * @return True if the password is changed
     */
    fun changeUserPassword(changePasswordDTO: ChangePasswordDTO): Boolean

    /**
     * Changes the password of a user using a code.
     *
     * @param changePasswordWithCodeDTO The DTO containing information to change the password.
     * @return True if password is changed
     */
    fun changeUserPasswordWithCode(changePasswordWithCodeDTO: ChangePasswordWithCodeDTO): Boolean
}